enumerable_methods.rb

```diff
@@ -0,0 +1,140 @@
+module Enumerable
+  # my each
+  def my_each
+    return to_enum unless block_given?
+    size.times do |n|
+      yield to_a[n]
+    end
+    self
+  end
+
+  # my_each_with_index
+  def my_each_with_index
+    to_enum unless block_given?
+    n = 0
+    while n < to_a.length
+      yield(to_a[n], n)
+      n += 1
+    end
+    self
+  end
+
+  # my_select
+  def my_select
+    return to_enum unless block_given?
+    selected = is_a?(Array || Range) ? [] : {}
+    if is_a?(Array)
+      my_each do |i|
+        selected.push(i) if yield(i)
+      end
+    else
+      my_each do |key, val|
+        selected[key] = val if yield(key, val)
+      end
+    end
+    selected
+  end
+
+  # my_all?
+  def my_all?(pattern = nil)
+    result = true
+    my_each do |i|
+      case pattern
+      when nil && !block_given?
+          result = false if !i
+      when Regexp
+        result = false unless i.match(pattern)
+      when Class
+        result = false unless i.is_a?(pattern)
+      when String, Numeric
+        result = false unless i == pattern
+      when nil
+        result = yield(i)
+      
+      end
+      break if result != true
+    end
+    p result
```

Comment:
You shouldn't be printing values 😆.

Reply:
The my_inject problem is due to the order of parameters. It was (start, sym), now it's (sym, start), and working